docs/modules/ROOT/pages/how-tos/exoscale/remove_storage_node.adoc

```diff
@@ -0,0 +1,243 @@
+
+== Remove Node
+Make a note of the node you want to remove
+
+[source,bash]
+----
+export NODE_TO_REMOVE=storage-XXXX
+----
+
+=== Remove old OSD
+. Make sure ArgoCD ran and reduced the target number of OSDs
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster \
+  get cephcluster cluster -o jsonpath='{.spec.storage.storageClassDeviceSets[0].count}'
+----
+
+:osd-node: ${NODE_TO_REMOVE}
+include::partial$storage-ceph-remove-osd.adoc[]
+
+. Make a note of the PVC of the old OSD
++
+NOTE: We also extract the name of the PV here, but we'll only delete the PV after removing the node from the cluster.
++
+[source,bash]
+----
+pvc_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+  "rook-ceph-osd-${OSD_ID}" -ojsonpath='{.metadata.labels.ceph\.rook\.io/pvc}')
+pv_name=$(kubectl --as=cluster-admin -n syn-rook-ceph-cluster get pvc \
+  "${pvc_name}" -o jsonpath='{.spec.volumeName}')
+----
+
+. Check if the OSD deployment needs to be deleted, and delete it if necessary
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster get deploy \
+  -l failure-domain="${NODE_TO_REMOVE}"
+# Run this command if the previous command lists a deployment
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete deploy \
+  -l failure-domain="${NODE_TO_REMOVE}"
+----
+
+. Clean up PVC and prepare job of the old OSD if necessary
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete job \
+  -l ceph.rook.io/pvc="${pvc_name}"
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete pvc "${pvc_name}"
+----
+
+. Clean up PVC encryption secret
++
+[source,bash]
+----
+kubectl --as=cluster-admin -n syn-rook-ceph-cluster delete secret -l pvc_name="${pvc_name}"
+----
+
+=== Remove the old MON
+
+:mon_node: ${NODE_TO_REMOVE}
+include::partial$storage-ceph-remove-mon.adoc[]
+
+=== Remove VM
+
+
+. Find Terraform resource index of the node to replace
++
+[source,bash]
+----
+# Grab JSON copy of current Terraform state
+terraform state pull > .tfstate.json
+node_index=$(jq --arg storage_node "${NODE_TO_REMOVE}" -r \
+  '.resources[] |
+   select(.module=="module.cluster.module.storage" and .type=="random_id") |
+   .instances[] |
+   select(.attributes.hex==$storage_node) |
+   .index_key' \
+  .tfstate.json)
+----
+
+. Verify that resource index is correct
++
+[source,bash]
+----
+jq --arg index "${node_index}" -r \
+  '.resources[] |
+   select(.module=="module.cluster.module.storage" and .type=="exoscale_compute") |
+   .instances[$index|tonumber] |
+   .attributes.hostname' \
+   .tfstate.json
+----
+
+. Remove node ID and node resource for node that we want to remove from the Terraform state
++
+[source,bash]
+----
+terraform state rm "module.cluster.module.storage.random_id.node_id[$node_index]"
+terraform state rm "module.cluster.module.storage.exoscale_compute.nodes[$node_index]"
+----
+
+:delete-node: ${NODE_TO_REMOVE}
+include::partial$exoscale/delete-node.adoc[]
```

Comment:
This will completely break subsequent runs of `terraform apply` if any storage node except the one with the highest index in the Terraform state is removed.

I'd suggest extracting the node to remove from the Terraform state (highest ID in the state) and adjust the how-to to remove that node. Deleting the VM on Exoscale then becomes

```
terraform apply
```